WcaOnRails/spec/controllers/teams_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+require 'rails_helper'
+
+describe TeamsController do
+  let(:team) { FactoryGirl.create :team }
+
+  describe "GET #index" do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :index
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the teams index page' do
+        get :index
+        expect(response).to render_template :index
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :index
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe "GET #new" do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :new
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the teams index page' do
+        get :new
+        expect(response).to render_template :new
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :new
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when not signed in' do
+      it 'redirects to the sign in page' do
+        post :create, team: { name: "Team2016" }
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+      it 'does not allow creation' do
+        post :create, team: { name: "Team2016" }
+        expect(response).to redirect_to root_url
+      end
+    end
+
+    context 'when signed in as an admin' do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'creates a new team' do
+        post :create, team: { name: "Team2016" }
+        new_team = Team.find_by_name("Team2016")
+        expect(response).to redirect_to edit_team_path(new_team)
+        expect(new_team.name).to eq "Team2016"
+      end
+    end
+  end
+
+  describe 'POST #update' do
```

Comment:
Please add a test that checks that you cannot set someone's start_date before their end_date, along with a correpsonding validation in TeamMember.